### PR TITLE
Story3 - added the claim a book function plus CSS

### PR DESCRIPTION
**Pull request by JaylWW**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -4,17 +4,18 @@ import Nav from './components/Nav/Nav'
...
-        <Route path='/' element={<Home />} />
-        <Route path='*' element={<Error />} />
-        <Route path='/books/:id' element={<SingleBookDetail />} />
-      </Routes>
+        <Routes>
+            <Route path='/' element={<Home />} />
+            <Route path='/books/claimed' element={<ClaimedBooks />} />
+            <Route path='/books/:id' element={<SingleBookDetail />} />
+            <Route path='*' element={<Error />} />
+        </Routes>
```
Comment (Geoff10) [inline src/App.jsx]: Tidy up the indentation here, please.

File: src/components/Booklisting/SingleBookDetail/SingleBookDetail.css [hunk 2]
```diff
@@ -34,4 +36,27 @@
...
+
+.display_container form {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-content: center;
+    align-items: center;
+    margin-top: 0px;
+    
+}
```
Comment (Geoff10) [inline src/components/Booklisting/SingleBookDetail/SingleBookDetail.css:49]: To make this consistent, a newline is needed here.

File: src/components/Booklisting/SingleBookDetail/SingleBookDetail.css [hunk 3]
```diff
@@ -34,4 +36,27 @@
...
+.claimedBookForm p{
```
Comment (Geoff10) [inline src/components/Booklisting/SingleBookDetail/SingleBookDetail.css]: For consistency, add a space after `p` before `{`.

File: src/components/Booklisting/SingleBookDetail/SingleBookDetail.jsx [hunk 4]
```diff
@@ -27,23 +30,41 @@ function SingleBookDetail () {
...
+                <div className="claimedBookForm">
+                    {/* if isClaimed == null, display the form to claim the book */}
+                    {!capitalName && <ClaimBookForm bookclaim={setCapitalName} id={id}/> }
+                    {capitalName && <p><strong>Claimed by:&nbsp;</strong> {capitalName}</p>}
+                </div> 
+        </div>
+        <div className="display_container form">
+                
+        </div>
         </div>
```
Comment (Geoff10) [inline src/components/Booklisting/SingleBookDetail/SingleBookDetail.jsx]: It looks like the indentation around here might need some adjustment.

File: src/components/ClaimBookForm/ClaimBookForm.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                console.log('fail')
+                // Error in Claiming -> **TODO** Red text at the top of the form saying error submitting. 
+                
+                // This response.json() call may come in handy later when we want to display some text. However it doesn't work, Geoff is investigating. 
+                
+                // response.json().then(function (response_json) {
+                //     console.log(response_json)
+                    
+
+                // })
```
Comment (Geoff10) [inline src/components/ClaimBookForm/ClaimBookForm.jsx]: Error handling needs to happen here so that the user sees why claiming the book didn't work. I can go through why this isn't currently working.

File: src/components/ClaimBookForm/ClaimBookForm.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
+import { useState } from "react"
+import "./ClaimBookForm.css"
+
+function ClaimBookForm({id, bookclaim}) {
+    
+    const [name, setName] = useState('')
+    const [email, setEmail] = useState('')
```
Comment (Geoff10) [inline src/components/ClaimBookForm/ClaimBookForm.jsx:7]: It looks like the state here isn't being used when sending the form, was there a reason for this?

File: src/components/Home/Home.jsx [hunk 7]
```diff
@@ -6,7 +6,7 @@ function Home() {
     const [books, setBooks] = useState([])
     
     useEffect (function() {
-        fetch('https://book-swap-api.dev.io-academy.uk/api/books')
+        fetch('https://book-swap-api.dev.io-academy.uk/api/books?claimed=0')
```
Comment (Geoff10) [inline src/components/Home/Home.jsx:9]: This is out of scope for this story.

File: src/components/ClaimedBooks/ClaimedBooks.jsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react"
```
Comment (Geoff10) [inline src/components/ClaimedBooks/ClaimedBooks.jsx:1]: This is out of scope for this story.

File: src/components/Nav/Nav.jsx [hunk 9]
```diff
@@ -7,7 +7,12 @@ function Nav() {
             <ul>
                 <NavLink to='/'>Book Swap</NavLink>
             </ul>
+            <ul className="navbooks">
+                <NavLink to="/books/claimed">Claimed Books</NavLink>
+                <NavLink to="/">Available Books</NavLink>
+            </ul>
```
Comment (Geoff10) [inline src/components/Nav/Nav.jsx:13]: This is out of scope for this story.

**Review by Geoff10 — CHANGES_REQUESTED** [review]
Jut a couple of things to look at.

File: src/components/ClaimedBooks/ClaimedBooks.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import { useParams } from "react-router-dom"
+import BookListing from "../Booklisting/Booklisting"
+
+function ClaimedBooks() {
+    const [books, setBooks] = useState ([])
+    const [clickedBookId, setclickedBookId] = useState(null)
```
Comment (Geoff10) [inline src/components/ClaimedBooks/ClaimedBooks.jsx]: This state isn't being used so it should be removed if it isn't needed.

File: src/components/ClaimedBooks/ClaimedBooks.jsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
```
Comment (Geoff10) [inline src/components/ClaimedBooks/ClaimedBooks.jsx]: This import isn't being used and should be removed.

File: src/components/ClaimedBooks/ClaimedBooks.jsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+            {books.map(book => 
+            <BookListing
+            title={book.title} 
+            author={book.author} 
+            image={book.image} 
+            genre={book.genre.name} 
+            id={book.id} 
+            key={book.id}
+            setClickedBookId={setclickedBookId}/>    
+            )}
```
Comment (Geoff10) [inline src/components/ClaimedBooks/ClaimedBooks.jsx]: The indentation can be tidied here for readability.

**Review by Geoff10 — CHANGES_REQUESTED** [review]
Jus a couple more things for the stuff that covers story 4.

File: src/components/ClaimedBooks/ClaimedBooks.jsx (hunk 10 above)
Comment (abdex) [reply src/components/ClaimedBooks/ClaimedBooks.jsx]: e6cf020

File: src/components/ClaimedBooks/ClaimedBooks.jsx (hunk 11 above)
Comment (abdex) [reply src/components/ClaimedBooks/ClaimedBooks.jsx]: e6cf020

File: src/components/ClaimedBooks/ClaimedBooks.jsx (hunk 12 above)
Comment (abdex) [reply src/components/ClaimedBooks/ClaimedBooks.jsx]: d8a5741

**Review by Geoff10 — APPROVED** [review]
Looks great!
